legacy_promotions/app/decorators/solidus_legacy_promotions/models/spree/order_decorator.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module SolidusLegacyPromotions
+  module OrderDecorator
+    def self.prepended(base)
+      base.state_machine do
+        if states[:delivery]
+          before_transition from: :delivery, do: :apply_shipping_promotions
```

Comment:
This could be formally breaking if someone replaced their whole state machine and removed this transition. But, I think it's acceptable because if that store doesn't use shipping promotions there won't be any change, and if they use it, it will just be recalculated twice.